code/game/machinery/doors/door.dm

```diff
@@ -76,8 +76,14 @@
 		var/mob/living/M = AM
 		if(world.time - M.last_bumped <= 10) return	//Can bump-open one airlock per second. This is to prevent shock spam.
 		M.last_bumped = world.time
-		if(!M.restrained() && M.mob_size > MOB_SIZE_SMALL)
-			bumpopen(M)
+		if(!M.restrained())
+			if(ispet(AM))
+				var/mob/living/simple_animal/A = AM
+				if(A.collar)

```

Comment:
@MarcellusPye `AM` is a `/mob/living` type, which doesn't have the collar var, so it won't compile if he tries `&& AM.collar`